cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -192,10 +211,27 @@ runQueryTip (AnyConsensusModeParams cModeParams) network mOutFile = do
     & fmap tuple3Fst
     & logExceptContinue renderShelleyQueryCmdError
 
+  tipTimeResult <- mProgressQuery anyEra consensusMode localNodeConnInfo tipSlotNo
+    <&> bimap toJsonPastHorizonException fst
+
+  let jsonTipTime :: Aeson.Value = either identity (toJSON . relativeTimeSeconds) tipTimeResult
+
+  systemStart <- mStartTime consensusMode localNodeConnInfo
```

Comment:
We ought to be able to consolodate all of these into a single connection to the node, which gets a consistent snapshot of the data. With separate connections each using the chain tip we do not guarantee that it's for the same tip.

Reply:
I made some steps towards this, but looking for tips on how I can push this further.

Reply:
This should be using one connection now.